app/src/pages/inside/common/launchFiltersContainer/launchFiltersContainer.jsx

```diff
@@ -117,8 +117,16 @@ export class LaunchFiltersContainer extends Component {
     const { launchesFiltersReady, launchFilters } = this.props;
 
     if (launchesFiltersReady) {
-      if (launchFilters !== prevProps.launchFilters || !this.state.isFilterTracked) {
-        this.trackFilters();
+      const launchFilterArrayStatistic = this.getFiltersCountStatistic(launchFilters);
+      const prevPropsLaunchFilterArrayStatistic = this.getFiltersCountStatistic(
+        prevProps.launchFilters,
+      );
+      if (
+        JSON.stringify(launchFilterArrayStatistic) !==
+          JSON.stringify(prevPropsLaunchFilterArrayStatistic) ||
+        !this.state.isFilterTracked
+      ) {
```

Comment:
Why not just?
```js
if (launchFilters.length !== prevProps.launchFilters.length || !this.state.isFilterTracked) {}
```
Previous solution looks much easier at the first glance

Reply:
Requirements changed. It is necessary to track saved and unsaved filters.

Reply:
I see. But in https://github.com/reportportal/service-ui/pull/2942/commits/14c3bc06b77b532cf1d298d2e2c925a094290c25 you track saved and unsaved filter like `0#0` what the difference? Also for now on every change add\remove\change conditions `this.getFiltersCountStatistic` is called